### PR TITLE
ITE drivers/timer: fix timer

### LINKED ISSUE #42847: it8xxx2_evb: Multiple tests in tests/subsys/portability/cmsis_rtos_v2 fail.
**Describe the bug**
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.semaphore
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.mempool
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.mempool_dynamic
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.messageq
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.event_flags_no_wait_timeout
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.event_flags_signalled
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.event_flags_isr
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.thread_flags_no_wait_timeout
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.thread_flags_signalled
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.thread_flags_isr
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.thread_join
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.thread_detached
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.thread_joinable_detach
portability.cmsis_rtos_v2 : portability.cmsis_rtos_v2.thread_joinable_terminate

**To Reproduce**
1. cd zephyrproject
2. ./zephyr/scripts/twister -p it8xxx2_evb --device-testing -T zephyr/tests/subsys/portability/cmsis_rtos_v2 --west-flash="../itetool/loader.sh" --device-serial=/dev/ttyUSB0
3. See error

**Logs and console output**
> ***** delaying boot 1ms (per build configuration) *****
*** Booting Zephyr OS build v3.0.0-rc3-2-g8889114e3ca9  (delayed boot 1ms) ***
Running test suite test_cmsis_v2_apis
===================================================================
START - test_kernel_apis
PASS - test_kernel_apis in 0.2 seconds
===================================================================
START - test_delay
PASS - test_delay in 0.3 seconds
===================================================================
START - test_thread_apis
PASS - test_thread_apis in 0.5 seconds
===================================================================
START - test_thread_apis_dynamic
PASS - test_thread_apis_dynamic in 0.5 seconds
===================================================================
START - test_thread_prio
PASS - test_thread_prio in 0.2 seconds
===================================================================
START - test_thread_prio_dynamic
PASS - test_thread_prio_dynamic in 0.2 seconds
===================================================================
START - test_timer
oneshot_callback (Timer 1) = 1
periodic_callback (Timer 2) = 1
periodic_callback (Timer 2) = 2
periodic_callback (Timer 2) = 3
periodic_callback (Timer 2) = 4
periodic_callback (Timer 2) = 5
PASS - test_timer in 0.40 seconds
===================================================================
START - test_mutex
PASS - test_mutex in 0.102 seconds
===================================================================
START - test_mutex_lock_timeout
PASS - test_mutex_lock_timeout in 0.3 seconds
===================================================================
START - test_semaphore
ASSERTION FAIL [thread->base.pended_on] @ WEST_TOPDIR/zephyr/kernel/sched.c:643
E: 
E:  mcause: 2, Illegal instruction
E:   mtval: 0
E:      a0: 80011b68    t0: 00001880
E:      a1: 00000283    t1: 60000000
E:      a2: 00f02705    t2: fffffff5
E:      a3: 00f02700    t3: 00000020
E:      a4: 00000020    t4: 00000006
E:      a5: 00000004    t5: 00dead00
E:      a6: a0000000    t6: 00000004
E:      a7: 0000a000
E:                      tp: aaaaaaaa
E:      ra: 80009762
E:    mepc: 800051f0
E: mstatus: 00001800
E: 
E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Current thread: 0x80101748 (Mutex_check)
E: Halting system

**Environment (please complete the following information):**
 - OS: Ubuntu
 - Toolchain zephyr-sdk-0.13.2
 - Commit 8889114e3ca9

### LINKED ISSUE #43513: it8xxx2_evb: tests/kernel/sleep failed
**Describe the bug**
The test: "tests/kernel/sleep" failed.

**To Reproduce**
1. source zephyr-env.sh
2. twister -p it8xxx2_evb --device-testing -T tests/kernel/sleep/ --west-flash="$HOME/itetool/loader.sh" --device-serial=/dev/ttyUSB0
3. See error

**Expected behavior**
The test should pass.

**Impact**
No serious impact.

**Logs and console output**
```
***** delaying boot 1ms (per build configuration) *****
*** Booting Zephyr OS build zephyr-v3.0.0-808-gd74cb2a3df23  (delayed boot 1ms) ***
Running test suite sleep
===================================================================
START - test_sleep
Kernel objects initialized
Test thread started: id = 0x801010f8
Helper thread started: id = 0x80101050
Testing normal expiration of k_sleep()
Testing: test thread sleep + helper thread wakeup test
Testing: test thread sleep + isr offload wakeup test
Testing: test thread sleep + main wakeup test thread
Testing kernel k_sleep()
PASS - test_sleep in 2.34 seconds
===================================================================
START - test_usleep
elapsed_ms = 5750
Assertion failed at WEST_TOPDIR/zephyr/tests/kernel/sleep/src/usleep.c:95: test_usleep: (elapsed_ms <= UPPER_BOUND_MS is false)
overslept
FAIL - test_usleep in 57.516 seconds
===================================================================
START - test_sleep_forever
Kernel objects initialized
PASS - test_sleep_forever in 0.4 seconds
===================================================================
Test suite sleep failed.
===================================================================
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: Ubuntu
 - Toolchain Zephyr SDK 0.13.2
 - Commit d74cb2a3df23

### PR DESCRIPTION
**Pull request by RuibinChang**
1.We don't need to convert the free run clock count,
that will be converted by the kernel
(base on CONFIG_SYS_CLOCK_TICKS_PER_SEC),
so we should return the HW register count value directly.

2.ITE RTOS timer HW frequency is fixed at 32768Hz, because this
clock source is always active in any EC mode (running/doze/deep doze).

3.When run the test_mutex_lock_timeout(), we need more time (>300us) for the testing 
thread to finish the job, or we will get a fail (we didn't run the test before). 

Because our event timer doesn't handle the float part, and 32768 is divisible by 8192 which
 is closest to kernel tick default 10000, I add CONFIG_SYS_CLOCK_TICKS_PER_SEC = 8192 
for it8xxx2_evb.

If the CONFIG_SYS_CLOCK_TICKS_PER_SEC = 32768, the 10 tick of timeout = 300us.
If the CONFIG_SYS_CLOCK_TICKS_PER_SEC = 8192, the 10 tick of timeout = 1200us. 
So we can get more time to finish the job.

fixes https://github.com/zephyrproject-rtos/zephyr/issues/43513
fixes https://github.com/zephyrproject-rtos/zephyr/issues/42847

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>

### CODE REVIEW

**Review by brockus-zephyr — CHANGES_REQUESTED** [review]
You added CONFIG_SYS_CLOCK_TICKS_PER_SEC definition but didn't describe why you need that now but didn't before.  Guessing the change to not convert system clock cycles to ticks in the other file is linked to this.

The commit message does not tell me anything about what is changing or why.  Please change the commit message to describe the change.

**Comment by RuibinChang** [comment]
> You added CONFIG_SYS_CLOCK_TICKS_PER_SEC definition but didn't describe why you need that now but didn't before. Guessing the change to not convert system clock cycles to ticks in the other file is linked to this.
> 
> The commit message does not tell me anything about what is changing or why. Please change the commit message to describe the change.

Done

**Review by brockus-zephyr — DISMISSED** [review]

**Comment by RuibinChang** [comment]
Update commit message (delete redundant space)

**Comment by RuibinChang** [comment]
Rebase the commits

**Review by brockus-zephyr — DISMISSED** [review]

File: boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig [hunk 1]
```diff
@@ -14,6 +14,7 @@ CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_UART_CONSOLE=y
 
 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=32768
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=8192
```
Comment (keith-zephyr) [inline boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig:17]: Should you just change the default setting defined [here](https://github.com/zephyrproject-rtos/zephyr/blob/6be45c2a1809fb7e2cf8919127d86c7c68df3c29/soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series#L28) instead?

Comment (RuibinChang) [reply boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig]: Done

**Review by fkokosinski — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by andyross — APPROVED** [review]

**Pull request merged**